### PR TITLE
Password special characters issue fix

### PR DESCRIPTION
**Pull request by Greengrer**

### CODE REVIEW

File: src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java [hunk 1]
```diff
@@ -41,10 +38,18 @@ void beforeAll() {
     }
 
     @Test
-    @DisplayName("Checks if the result of hashing with 10000 iterations is the same for equal inputs.")
+    @DisplayName("Checkes if the result of hashing with 10000 iterations is the same for equal inputs.")
     void hashPasswordWithTenThousandIterationsTest() {
         String resultHash = securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000);
-        assertTrue(encoder.matches(passwordToHash, securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000)));
+        assertEquals(resultHash,
+                securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000));
```
Comment (hemiSRT8) [inline src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java]: one line

File: src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java [hunk 2]
```diff
@@ -41,10 +38,18 @@ void beforeAll() {
     }
 
     @Test
-    @DisplayName("Checks if the result of hashing with 10000 iterations is the same for equal inputs.")
+    @DisplayName("Checkes if the result of hashing with 10000 iterations is the same for equal inputs.")
```
Comment (hemiSRT8) [inline src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java]: Checks?

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java [hunk 3]
```diff
@@ -79,7 +79,7 @@ public void emailFormatCheck(String email) {
 
     @Override
     public void validatePassword(String password) {
-        if (!password.matches("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)[a-zA-Z\\d]{8,}$")) {
+        if (!password.matches("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>\\/?]{8,}$")) {
```
Comment (shelsoftensy) [inline src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java:82]: maybe move string to a constant value?

Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java]: maybe move string to a constant value?

**Review by littlepony99 — CHANGES_REQUESTED** [review]

File: src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java [hunk 4]
```diff
@@ -20,6 +20,8 @@
     private final Random random = new SecureRandom();
     private final SecretKeyFactory secretKeyFactory;
     private PasswordEncoder encoder;
+    private final String passwordValidator = "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>\\/?]{8,}$";
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java]: make it static final and name it PASSWORD_PATTERN

File: src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java [hunk 5]
```diff
@@ -20,6 +20,8 @@
...
+    private final String emailValidator = "/^[^\\s@]+@[^\\s@]+$/";
```
Comment (hemiSRT8) [inline src/main/java/com/vinylteam/vinyl/security/impl/DefaultSecurityService.java]: make it static final and name it EMAIL_PATTERN

File: src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java [hunk 6]
```diff
@@ -44,7 +41,14 @@ void beforeAll() {
...
     void hashPasswordWithTenThousandIterationsTest() {
         String resultHash = securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000);
-        assertTrue(encoder.matches(passwordToHash, securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000)));
+        assertEquals(resultHash, securityService.hashPassword(passwordToHash.toCharArray(), salt, 10000));
+    }
+
+    @Test
+    @DisplayName("Checks if hashing with zero iterations throws an IllegalArgumentException.")
+    void hashPasswordWithZeroIterationsTest() {
+        assertThrows(IllegalArgumentException.class, () ->
```
Comment (hemiSRT8) [inline src/test/java/com/vinylteam/vinyl/security/impl/DefaultSecurityServiceTest.java:50]: one line

**Review by hemiSRT8 — CHANGES_REQUESTED** [review]

**Review by littlepony99 — APPROVED** [review]

**Review by hemiSRT8 — APPROVED** [review]

**Pull request merged**